domain/runaway.go

```diff
@@ -176,8 +176,18 @@ func (do *Domain) runawayWatchSyncLoop() {
 }
 
 // AddRunawayWatch is used to add runaway watch item manually.
-func (do *Domain) AddRunawayWatch(record *resourcegroup.QuarantineRecord) error {
-	return do.handleRunawayWatch(record)
+func (do *Domain) AddRunawayWatch(record *resourcegroup.QuarantineRecord) (uint64, error) {
+	if err := do.handleRunawayWatch(record); err != nil {
+		return 0, err
+	}
+	if err := do.TryToUpdateRunawayWatch(); err != nil {
+		return 0, err
+	}
+	r := do.runawayManager.GetWatchByKey(record.GetRecordKey())
+	if r != nil {
```

Comment:
Here goes the same issue as before:
- If `handleRunawayWatch` return success, it means the record is inserted successfully, so it's like runaway manager can see this record within one update interval. But It's possible `TryToUpdateRunawayWatch` failed due to arbitrary errors(e.g. network error), thus the function returns an error but the record is actually takes effect.
- In another case, since table `tidb_runaway_watch`'s primary key is the auto_increment id, it's possible that two concurrent insert with the same recordkey can overwrite each other when put into the runaway manager, so user `select LAST_INSERT_ID()` here to get the id may be the better choice.

Reply:
Get it. I think we can retry when `select LAST_INSERT_ID()` fails and tell the user that the insertion is successful but getting id fails.